smack-experimental/src/main/java/org/jivesoftware/smackx/chat_markers/ChatMarkersManager.java

```diff
@@ -70,6 +109,76 @@ public static synchronized ChatMarkersManager getInstanceFor(XMPPConnection conn
 
     private ChatMarkersManager(XMPPConnection connection) {
         super(connection);
+        connection.addStanzaInterceptor(new StanzaListener() {
+            @Override
+            public void processStanza(Stanza packet)
+                    throws
+                    NotConnectedException,
+                    InterruptedException,
+                    SmackException.NotLoggedInException {
+                Message message = (Message) packet;
+                if (!isEligibleForChatMarkers(message)) {
+                    return;
+                }
+
+                if (!MessageWithBodiesFilter.INSTANCE.accept(message)) {
```

Comment:
Don't use the filter within the listener. Use it with the filter argument of addStanzaInterceptor(), e.g.:

```
new AndFilter(MessageTypeFilter.NORMAL_OR_CHAT_OR_GROUPCHAT, MessageWithBodiesFilter.INSTANCE, ChatMarkersFilter.INSTANCE)
```

Reply:
ok, done, I created:
```
private static final StanzaFilter OUTGOING_MESSAGE_FILTER = new AndFilter(
            MessageTypeFilter.NORMAL_OR_CHAT_OR_GROUPCHAT,
            MessageWithBodiesFilter.INSTANCE,
            new NotFilter(ChatMarkersFilter.INSTANCE)
    );
```
and used it in filter argument of addStanzaInterceptor()